edi_bank_statement_import/tests/edi_bank_statement_import_test.py

```diff
@@ -0,0 +1,161 @@
+                "partner_id": self.env.ref("base.main_partner").id,
+                "company_id": self.env.ref("base.main_company").id,
+                "bank_id": self.env.ref("base.res_bank_1").id,
+            }
+        )
+        self.env["res.partner.bank"].create(
+            {
+                "acc_number": "NL46ABNA0499998748",
+                "partner_id": self.env.ref("base.main_partner").id,
+                "company_id": self.env.ref("base.main_company").id,
+                "bank_id": self.env.ref("base.res_bank_1").id,
+            }
+        )
+        self.env["account.journal"].create(
+            {
+                "name": "Bank Journal - (test camt)",
+                "code": "TBNKCAMT",
+                "type": "bank",
+                "bank_account_id": bank.id,
+                "currency_id": self.env.ref("base.EUR").id,
+            }
+        )
+
+    def test_statement_import(self):
+        """Test correct creation of single statement."""
+        testfile = get_module_resource(
+            "account_statement_import_camt", "test_files", "test-camt053"
+        )
+        with open(testfile, "rb") as datafile:
+            camt_file = base64.b64encode(datafile.read())
+
+            self.env["account.statement.import"].create(
+                {
+                    "statement_filename": "test import",
+                    "statement_file": camt_file,
+                }
+            ).import_file_button()
+
+            bank_st_record = self.env["account.bank.statement"].search(
+                [("name", "=", "1234Test/1")], limit=1
+            )
+            statement_lines = bank_st_record.line_ids
+            self.assertTrue(
+                any(
+                    all(
+                        line[key] == self.transactions[0][key]
+                        for key in ["amount", "date", "ref"]
+                    )
+                    # TODO and bank_account_id was removed from line
+                    # and line.bank_account_id.acc_number
+                    # == self.transactions[0]["account_number"]
+                    for line in statement_lines
+                )
+            )
+
+    def test_zip_import(self):
+        """Test import of multiple statements from zip file."""
+        testfile = get_module_resource(
+            "account_statement_import_camt", "test_files", "test-camt053.zip"
+        )
+        with open(testfile, "rb") as datafile:
+            camt_file = base64.b64encode(datafile.read())
+            self.env["account.statement.import"].create(
+                {"statement_filename": "test import", "statement_file": camt_file}
+            ).import_file_button()
+            bank_st_record = self.env["account.bank.statement"].search(
+                [("name", "in", ["1234Test/2", "1234Test/3"])]
+            )
+
+        self.assertTrue(all([st.line_ids for st in bank_st_record]))
+        self.assertEqual(bank_st_record[0].line_ids.mapped("sequence"), [1, 2, 3])
```

Comment:
pls have a look at how tests are done in edi_oca and test the component.

Reply:
removed the tests